tests/integration/targets/aci_vrf_leak_internal_subnet/tasks/main.yml

```diff
@@ -0,0 +1,208 @@
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when:
+  - query_cloud.current == []  # This condition will execute only non-cloud sites
+  - version.current.0.topSystem.attributes.version is version('5', '>=')
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Set vars
+    set_fact:
+      aci_info: &aci_info
+        host: '{{ aci_hostname }}'
+        username: '{{ aci_username }}'
+        password: '{{ aci_password }}'
+        validate_certs: '{{ aci_validate_certs | default(false) }}'
+        use_ssl: '{{ aci_use_ssl | default(true) }}'
+        use_proxy: '{{ aci_use_proxy | default(true) }}'
+        output_level: '{{ aci_output_level | default("debug") }}'
+
+  - name: delete existing tenant
+    cisco.aci.aci_tenant: 
+      <<: *aci_info
+      state: absent
+      tenant: ansible_test3
+    register: tenant_present
+
+  - name: ensure tenant exists for tests to kick off
+    cisco.aci.aci_tenant: &aci_tenant_present
+      <<: *aci_info
+      state: present
+      tenant: ansible_test3
+    register: tenant_present
+
+  - name: create vrf 
+    cisco.aci.aci_vrf: &aci_vrf_present
+      <<: *aci_tenant_present
+      vrf: test
+      description: Ansible Test
+    register: vrf_present
+
+  - name: create second vrf - creation works
+    cisco.aci.aci_vrf:
+      <<: *aci_vrf_present
+      vrf: test2
+    register: vrf_present
+
+  - name: create third vrf - creation works
+    cisco.aci.aci_vrf:
+      <<: *aci_vrf_present
+      vrf: test3
+    register: vrf_present
+
+  - name: create leak internal subnet - check_mode
+    cisco.aci.aci_vrf_leak_internal_subnet: &aci_leak_internal_subnet
+      <<: *aci_info
+      state: present
+      tenant: ansible_test3
+      vrf: test
+      ip: 1.1.1.2
+      leak_to: 
+        - vrf: "test2"
+          tenant: "ansible_test3"
+        - vrf: "test3"
+          tenant: "ansible_test3"
+      description: Ansible Test
+    check_mode: true
+    register: leak_sub_check_mode
+
+  - name: create leak internal subnet 
+    cisco.aci.aci_vrf_leak_internal_subnet:
+      <<: *aci_leak_internal_subnet
+    register: leak_sub_present
+
+  - name: create leak internal subnet - idempotency check
+    cisco.aci.aci_vrf_leak_internal_subnet: 
+      <<: *aci_info
+      state: present
+      tenant: ansible_test3
+      vrf: test
+      ip: 1.1.1.2
+      leak_to: 
+        - vrf: "test2"
+          tenant: "ansible_test3"
+        - vrf: "test3"
+          tenant: "ansible_test3"
+      description: Ansible Test
+    register: leak_sub_idempotent
+
+  - name: create second leak internal subnet
+    cisco.aci.aci_vrf_leak_internal_subnet: 
+      <<: *aci_info
+      tenant: ansible_test3
+      vrf: test2
+      leak_to: 
+        - vrf: "test"
+          tenant: "ansible_test3"
+      description: Ansible Test
+      ip: 1.1.1.2
+    register: leak_sub_present_2
+
```

Comment:
It would be good to sort the saved output in **leak_sub_present** first and then run the assert tests. The child objects are not returned in the order they were sent/created.
Example -
```
leak_to: 
        - vrf: "test2"
          tenant: "ansible_test3"
        - vrf: "test3"
          tenant: "ansible_test3"
```
can be returned via GET as either -
```
"children": [
                    {
                        "leakTo": {
                            "attributes": {
                                "annotation": "",
                                "ctxName": "test3",
                                "descr": "",
                                "name": "",
                                "nameAlias": "",
                                "scope": "private",
                                "tenantName": "ansible_test3",
                                "userdom": ":all:common:"
                            }
                        }
                    },
                    {
                        "leakTo": {
                            "attributes": {
                                "annotation": "",
                                "ctxName": "test2",
                                "descr": "",
                                "name": "",
                                "nameAlias": "",
                                "scope": "private",
                                "tenantName": "ansible_test3",
                                "userdom": ":all:common:"
                            }
                        }
                    }
                ]
```
or -
```
"children": [
                    {
                        "leakTo": {
                            "attributes": {
                                "annotation": "",
                                "ctxName": "test2",
                                "descr": "",
                                "name": "",
                                "nameAlias": "",
                                "scope": "private",
                                "tenantName": "ansible_test3",
                                "userdom": ":all:common:"
                            }
                        }
                    },
                    {
                        "leakTo": {
                            "attributes": {
                                "annotation": "",
                                "ctxName": "test3",
                                "descr": "",
                                "name": "",
                                "nameAlias": "",
                                "scope": "private",
                                "tenantName": "ansible_test3",
                                "userdom": ":all:common:"
                            }
                        }
                    }
                ]

```
You can replace the tasks with the below snippet-
```
- name: Sort the list of leaked internal subnets for present
    ansible.builtin.set_fact:
      attributes_list_present: "{{ leak_sub_present.current.0.leakInternalSubnet.children | map(attribute='leakTo.attributes') | list | sort(attribute='ctxName') }}"

  - name: present asserts
    assert:
      that:
        - vrf_present is changed
        - leak_sub_check_mode is changed
        - leak_sub_check_mode.proposed.leakInternalSubnet.attributes.ip == '1.1.1.2'
        - leak_sub_check_mode.proposed.leakInternalSubnet.children.0.leakTo.attributes.ctxName == 'test2'
        - leak_sub_check_mode.proposed.leakInternalSubnet.children.0.leakTo.attributes.tenantName == 'ansible_test3'
        - leak_sub_check_mode.proposed.leakInternalSubnet.children.1.leakTo.attributes.ctxName == 'test3'
        - leak_sub_check_mode.proposed.leakInternalSubnet.children.1.leakTo.attributes.tenantName == 'ansible_test3'
        - leak_sub_present.current.0.leakInternalSubnet.attributes.ip == '1.1.1.2'
        - attributes_list_present.0.tenantName == 'ansible_test3'
        - attributes_list_present.0.ctxName == 'test2'
        - attributes_list_present.1.tenantName == 'ansible_test3'
        - attributes_list_present.1.ctxName == 'test3'
        - leak_sub_idempotent is not changed
        - leak_sub_present_2.current.0.leakInternalSubnet.attributes.ip == '1.1.1.2'
        - leak_sub_present_2.current.0.leakInternalSubnet.children.0.leakTo.attributes.ctxName == 'test'
        - leak_sub_present_2.current.0.leakInternalSubnet.children.0.leakTo.attributes.tenantName == 'ansible_test3'
```